### PR TITLE
Feat meshes inspector

### PR DESCRIPTION
**Pull request by francoisbonneau**

### CODE REVIEW

File: include/geode/inspector/criterion/adjacency/brep_meshes_adjacency.h [hunk 1]
```diff
@@ -26,19 +26,30 @@
...
-    struct uuid;
-    struct PolygonEdge;
-    struct PolyhedronFacet;
 } // namespace geode
 
 namespace geode
 {
+    struct BRepMeshesAdjacencyInspectionResult
+    {
+        InspectionIssues< uuid > meshes_with_wrong_adjacencies{ "" };
```
Comment (BotellaA) [inline include/geode/inspector/criterion/adjacency/brep_meshes_adjacency.h:46]: Should we need a clear message?

Comment (BotellaA) [reply include/geode/inspector/criterion/adjacency/brep_meshes_adjacency.h:46]: here and everywhere else?

File: include/geode/inspector/criterion/colocation/edgedcurve_colocation.h [hunk 2]
```diff
@@ -45,16 +46,12 @@ namespace geode
...
-            const EdgedCurve< dimension >& mesh, bool verbose );
-
         ~EdgedCurveColocation();
 
         bool mesh_has_colocated_points() const;
 
-        index_t nb_colocated_points() const;
-
-        std::vector< std::vector< index_t > > colocated_points_groups() const;
+        InspectionIssues< std::vector< index_t > >
```
Comment (BotellaA) [inline include/geode/inspector/criterion/colocation/edgedcurve_colocation.h:53]: I am not sure about these outputs should we have a message to display or not? When do we need this kind of category message?

File: src/geode/inspector/criterion/manifold/brep_meshes_manifold.cpp [hunk 3]
```diff
@@ -46,124 +46,71 @@ namespace geode
...
                     {
-                        non_manifold_components.push_back( component_id );
+                        non_manifold_components.add_problem(
+                            model_edge_issues.second.problems[issue_id],
+                            model_edge_issues.second.messages[issue_id] );
                     }
                 }
             }
-            sort_unique( non_manifold_components );
+            // I commented that.... How to deal with it?
```
[the remark this answers is not in the document]
Comment (BotellaA) [reply src/geode/inspector/criterion/manifold/brep_meshes_manifold.cpp:99]: maybe just a sort_unique on the block.id() should be ok.

File: src/geode/inspector/topology/brep_topology.cpp [hunk 4]
```diff
@@ -80,48 +75,19 @@ namespace
...
-        }
+        Impl( const BRep& brep ) : brep_( brep ) {}
 
         bool brep_meshed_components_are_linked_to_unique_vertices() const
         {
             for( const auto& corner : brep_.corners() )
-            {
+            { // why not testing that the corner is meshed?
                 if( !brep_component_vertices_are_associated_to_unique_vertices(
                         brep_, corner.component_id(), corner.mesh() ) )
```
Comment (MelchiorSchuh) [inline src/geode/inspector/topology/brep_topology.cpp:92]: Add test to check if corner is meshed

File: src/geode/inspector/topology/brep_topology.cpp [hunk 5]
```diff
@@ -170,123 +136,92 @@ namespace geode
...
-                        Logger::info( "Unique vertex with id ", uv_id,
-                            " is not linked to any component mesh vertex." );
-                    }
+                    return false;
                 }
             }
-            return unlinked_uv;
+            return true;
+        }
+        BRepTopologyInspectionResult inspect_brep_topology(
```
Comment (MelchiorSchuh) [inline src/geode/inspector/topology/brep_topology.cpp:188]:
```suggestion
        }
        
        BRepTopologyInspectionResult inspect_brep_topology(
```

File: src/geode/inspector/criterion/manifold/brep_meshes_manifold.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (francoisbonneau) [reply src/geode/inspector/criterion/manifold/brep_meshes_manifold.cpp:99]: a sort unique on block id will remove some component of the Inspectorissue without removing the associated message.

File: include/geode/inspector/criterion/adjacency/brep_meshes_adjacency.h (hunk 1 above)
Comment (francoisbonneau) [reply include/geode/inspector/criterion/adjacency/brep_meshes_adjacency.h:46]: The message will be created in the inspection method

File: include/geode/inspector/criterion/colocation/edgedcurve_colocation.h (hunk 2 above)
Comment (francoisbonneau) [reply include/geode/inspector/criterion/colocation/edgedcurve_colocation.h:53]: this function feed the big struct.

File: include/geode/inspector/criterion/manifold/section_meshes_manifold.h [hunk 6]
```diff
@@ -26,17 +26,26 @@
...
+    struct SectionMeshesManifoldInspectionResult
+    {
+        InspectionIssues< uuid > non_manifold_meshes{
+            "Non manifold section meshes"
+        };
+        absl::flat_hash_map< uuid, InspectionIssues< index_t > >
+            meshes_non_manifold_vertices;
+        absl::flat_hash_map< uuid,
+            InspectionIssues< std::array< index_t, 2 > > >
+            meshes_non_manifold_edges;
```
[the remark this answers is not in the document]
Comment (francoisbonneau) [reply include/geode/inspector/criterion/manifold/section_meshes_manifold.h:47]: je n ai pas enlevé de fonctionnalités.
